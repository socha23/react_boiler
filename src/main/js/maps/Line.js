import React from 'react';

export const STYLE_DEFAULT = {
    className: "",
    style: {
        height: 5,
        backgroundColor: "red",
        borderRadius: 2,
        transformOrigin: "2px 2px"
    }
};

function animatedStyle(className) {
    return {
        className: "progress-bar " + className + "  progress-bar-striped active",
        style: {
            height: 10,
            borderRadius: 5,
            animationDirection: "reverse",
            transformOrigin: "0 4px",
            transition: 'none',
            opacity: 0.8
        }
    };
}

export const STYLE_ANIMATED_RED = animatedStyle("progress-bar-danger");
export const STYLE_ANIMATED_GREEN = animatedStyle("progress-bar-success");
export const STYLE_ANIMATED_YELLOW = animatedStyle("progress-bar-warning");


const Line = ({fromX, fromY, toX, toY, style=STYLE_DEFAULT}) => {
    let dX = toX - fromX;
    let dY = toY - fromY;
    let len = Math.sqrt(dX * dX + dY * dY);

    let angle = Math.PI - Math.atan2(dY, -dX);


    return <div><div className={style.className} style={{
        ...style.style,
        position: "absolute",
        width: len,
        top: fromY,
        left: fromX,
        transform: "rotate(" + angle + "rad)"
    }}/></div>
};

export default Line;