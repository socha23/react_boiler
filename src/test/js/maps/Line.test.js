import React from 'react'
import renderer from 'react-test-renderer'

import Line from 'maps/Line'

it('renders correctly when straight', testSnapshot(
            <Line fromX={0} fromY={0} toX={100} toY={0}/>
));

it('renders correctly when angled', testSnapshot(
            <Line fromX={0} fromY={0} toX={100} toY={50}/>
));


function testSnapshot(elem) {
    return () => {
        expect(renderer.create(elem).toJSON()).toMatchSnapshot();
    }
}