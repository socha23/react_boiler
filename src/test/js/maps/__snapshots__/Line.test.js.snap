// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when angled 1`] = `
<div>
  <div
    className=""
    style={
      Object {
        "backgroundColor": "red",
        "borderRadius": 2,
        "height": 5,
        "left": 0,
        "position": "absolute",
        "top": 0,
        "transform": "rotate(0.46364760900080615rad)",
        "transformOrigin": "2px 2px",
        "width": 111.80339887498948,
      }
    }
  />
</div>
`;

exports[`renders correctly when straight 1`] = `
<div>
  <div
    className=""
    style={
      Object {
        "backgroundColor": "red",
        "borderRadius": 2,
        "height": 5,
        "left": 0,
        "position": "absolute",
        "top": 0,
        "transform": "rotate(0rad)",
        "transformOrigin": "2px 2px",
        "width": 100,
      }
    }
  />
</div>
`;
